### ISSUE TITLE
about the code of top-k filtering

### ISSUE THREAD

**Issue by mmyjjl1009** [hkchengrex/MiVOS#21]
thanks for your sharing.
what's the meaning of  
`x_exp = torch.exp(values - values[:, 0])` ?
why need to do values -values[:0]?
Looking forward to your reply.

**Comment by hkchengrex** [comment]
That is normalization for numerical stability. See also https://discuss.pytorch.org/t/how-to-implement-the-exactly-same-softmax-as-f-softmax-by-pytorch/44263
